sources/packages/backend/apps/api/src/route-controllers/student-appeal/_tests_/e2e/student-appeal.students.controller.getStudentAppealSummary.e2e-spec.ts

```diff
@@ -0,0 +1,241 @@
+      // Create an approved student application appeal for the student
+      // with 2 appeal requests.
+      const studentApplicationAppeal = await saveFakeAppealWithAppealRequests(
+        student,
+        [
+          {
+            appealStatus: StudentAppealStatus.Approved,
+            assessedDate,
+          },
+          {
+            appealStatus: StudentAppealStatus.Declined,
+            assessedDate,
+          },
+        ],
+        { isApplicationAppeal: true, programYear: eligibleProgramYear },
+      );
+      const [
+        studentApplicationAppealRequest1,
+        studentApplicationAppealRequest2,
+      ] = studentApplicationAppeal.appealRequests;
+      const application = studentApplicationAppeal.application;
+      // Create a legacy change request appeal that should not be returned in the summary.
+      await saveFakeAppealWithAppealRequests(
+        student,
+        [
+          {
+            appealStatus: StudentAppealStatus.Approved,
+          },
+        ],
+        { isApplicationAppeal: true },
+      );
+      const studentToken = await getStudentToken(
+        FakeStudentUsersTypes.FakeStudentUserType1,
+      );
+      await mockJWTUserInfo(appModule, student.user);
+
+      // Act/Assert
+      await request(app.getHttpServer())
+        .get(endpoint)
+        .auth(studentToken, BEARER_AUTH_TYPE)
+        .expect(HttpStatus.OK)
+        .expect({
+          appeals: [
+            {
+              id: studentApplicationAppeal.id,
+              appealStatus: StudentAppealStatus.Approved,
+              appealRequestNames: [
+                studentApplicationAppealRequest1.submittedFormName,
+                studentApplicationAppealRequest2.submittedFormName,
+              ],
+              applicationId: application.id,
+              applicationNumber: application.applicationNumber,
+              assessedDate: assessedDate.toISOString(),
+            },
+          ],
+        });
+    },
+  );
+
+  it("Should return empty list when the student has no appeals.", async () => {
+    // Arrange
+    const student = await saveFakeStudent(db.dataSource);
+    const studentToken = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    await mockJWTUserInfo(appModule, student.user);
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(studentToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        appeals: [],
+      });
+  });
+
+  /**
+   *  Saves a fake student appeal with the given appeal request inputs for the provided student.
+   * @param student student.
+   * @param appealRequestValues appeal request values.
+   * @param options options.
+   * - `isApplicationAppeal` indicates if the appeal is for an application.
+   * - `programYear` application program year.
+   * @returns the saved student appeal.
+   */
+  async function saveFakeAppealWithAppealRequests(
```

Comment:
The method seems generic enough to be moved to the factories. Was there a reason to keep it in file scope only?

Reply:
Moved the method to factory.